variables.tf

```diff
@@ -24,6 +24,10 @@ variable "project_id" {
   type        = string
   default     = ""
   description = "A project ID different from the default defined inside the provider"
+  validation {
```

Comment:
It would be great to have this validation enforced in the provider `lacework_integration_gcp_at` resource, which would allow some better unit tests.

In the `lacework_integration_gcp_at` resource if the resource level is PROJECT then resource_id is the project_id. Which would allow the regex validation on the project_id input.

Reply:
definitely makes sense to do this in our provider as well  but in our case if you are using GCP project number instead of project id our modules fails to set up the IAM permissions at https://github.com/lacework/terraform-gcp-audit-log/blob/d00443adaf2130a756256ff8ca6511b67650ab12/main.tf#L43-L45.